src/transformers/agents/agent_types.py

```diff
@@ -129,7 +129,7 @@ def to_raw(self):
             return self._raw
 
         if self._tensor is not None:
-            array = self._tensor.cpu().detach().numpy()
+            array = self._tensor.detach().numpy()
```

Comment:
`numpy()` behaves differently depending on whether it is on CPU or not, so we want to keep the original here ([docs](https://pytorch.org/docs/stable/generated/torch.Tensor.numpy.html#torch.Tensor.numpy))

cases with `.cpu()` and `.numpy()` should be kept as is -- there are many examples of this changes in this PR

Reply:
Reverted